test/tests/components/ws-manager/additional_repositories_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package wsmanager
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	csapi "github.com/gitpod-io/gitpod/content-service/api"
+	"github.com/gitpod-io/gitpod/test/pkg/integration"
+	wsmanapi "github.com/gitpod-io/gitpod/ws-manager/api"
+)
+
+func TestAdditionalRepositories(t *testing.T) {
+	f := features.New("additional-repositories").
+		WithLabel("component", "ws-manager").
+		Assess("can open a workspace using the additionalRepositories property", func(_ context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			tests := []struct {
+				Name       string
+				ContextURL string
+				CloneTaget string
+			}{
+				{
+					Name:       "workspace with additionalRepositories using a branch",
+					ContextURL: "https://github.com/gitpod-io/gitpod-test-repo",
+					CloneTaget: "aledbf/test-additional-repositories",
+				},
+				{
+					Name:       "workspace with additionalRepositories also using a branch in one of the additionalRepositories",
+					ContextURL: "https://github.com/gitpod-io/gitpod-test-repo",
+					CloneTaget: "aledbf/test-additional-repositories-with-branches",
+				},
+			}
+			ctx, cancel := context.WithTimeout(context.Background(), time.Duration(5*len(tests))*time.Minute)
+			defer cancel()
+
+			for _, test := range tests {
+				t.Run(test.Name, func(t *testing.T) {
+					t.Parallel()
```

Comment:
Limit concurrency. Otherwise, we will eventually fail with Too Many Requests.
https://github.com/gitpod-io/gitpod/blob/18c147fe4d92337d61161900fb029a67843fcf83/test/tests/workspace/contexts_test.go#L110

Reply:
removed